adr/2025_06_13-packages.md

```diff
@@ -0,0 +1,30 @@
+# BlockNote Packages & Sub-Packages
+
+## Core
+
+The `@blocknote/core` package should contain everything needed to build a pure JS editor instance. It has no opinion on the UI layer, and is only concerned with the core editor functionality.
+
+It contains sub-packages for:
+
+- `@blocknote/core/blocks` - Contains the default blocks, inline content, styles and their implementations (excluding things like blockContainer, blockGroup, etc which are core to the editor but should not be part of the schema).
+  - `@blocknote/core/blocks/header` - Contains the header block.
+  - `@blocknote/core/blocks/code-block` - Contains the code block block.
```

Comment:
except the code-block will probably stay in a different package right?

Reply:
There is the `code-block` default block, and there is the code-block package which would be for things like the shiki languages and all that. Probably worthwhile to revisit the naming there.